drivers/s3.go

```diff
@@ -392,6 +395,24 @@ func (os *s3Session) DeleteFile(ctx context.Context, name string) error {
 	return err
 }
 
+var wrongRegionRe = regexp.MustCompile("^AuthorizationHeaderMalformed: The authorization header is malformed; the region '([a-zA-Z0-9-]+)' is wrong; expecting '([a-zA-Z0-9-]+)'")
+
+// Given an error, attempt to adjust our configuration to facilitate a succesful retry. Returns true if we should retry.
+func (os *s3Session) recoverError(err error) bool {
+	results := wrongRegionRe.FindStringSubmatch(fmt.Sprint(err))
+	if len(results) == 0 {
+		return false
+	}
+	newRegion := results[2]
+	oldRegion := *os.s3sess.Config.Region
+	// Avoid an infinite loop; if nothing's going to change don't bother
+	if oldRegion == newRegion {
+		return false
+	}
+	os.s3sess.Config.Region = aws.String(newRegion)
```

Comment:
Is it safe to mutate the internal field of the session like this as well? May be safer to create a new session instead, WDYT? Sth like this might work:
```suggestion
	os.s3sess, err = session.NewSession(os.s3sess.Config.WithRegion(newRegion))
	if err != nil {
	    // log
  	    return false
	}
```

This will also avoid changing the internal field from the AWS SDK in the middle of other concurrent operations going on (not only the concurrency of this function)